monai/transforms/croppad/array.py

```
@@ -379,31 +388,25 @@ def compute_slices(
-                roi_center_t = convert_to_tensor(data=roi_center, dtype=torch.int16, wrap_sequence=True, device="cpu")
-                roi_size_t = convert_to_tensor(data=roi_size, dtype=torch.int16, wrap_sequence=True, device="cpu")
-                _zeros = torch.zeros_like(roi_center_t)
-                half = torch.divide(roi_size_t, 2, rounding_mode="floor")
-                roi_start_t = torch.maximum(roi_center_t - half, _zeros)
-                roi_end_t = torch.maximum(roi_start_t + roi_size_t, roi_start_t)
+                centers = _to_int_list(roi_center)
+                sizes = _to_int_list(roi_size)
+                n = max(len(centers), len(sizes))
+                centers = centers * n if len(centers) == 1 else centers
+                sizes = sizes * n if len(sizes) == 1 else sizes
+                starts = [max(c - s // 2, 0) for c, s in zip(centers, sizes)]
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)